system/Validation/Validation.php

```diff
@@ -206,8 +206,13 @@ public function check($value, string $rule, array $errors = []): bool
 	 *
 	 * @return boolean
 	 */
-	protected function processRules(string $field, string $label = null, $value, $rules = null, array $data): bool
+	protected function processRules(string $field, string $label = null, $value, $rules = null, array $data = null): bool
 	{
+		if (is_null($data))
+		{
+			throw new InvalidArgumentException('You must supply the parameter: data.');
+		}
+
```

Comment:
```suggestion
	protected function processRules(string $field, string $label = null, $value, $rules = null, array $data = []): bool
	{
		if ($data === [])
		{
			throw new InvalidArgumentException('You must supply the parameter: data.'); // @codeCoverageIgnore
		}

```

Reply:
This one has to add `null` as an input option because `[]` is actually valid input and we're trying to catch cases where the parameter was not supplied at all.